src/lib.rs

```diff
@@ -1740,6 +1755,73 @@ impl LightningNode {
             .status)
     }
 
+    /// Prepares a reverse swap that sends all funds in LN channels. This is possible because the
+    /// route to the swap service is known, so fees can be known in advance.
+    ///
+    /// The return includes fee estimates and must be provided to [`LightningNode::clear_wallet`] in
+    /// order to execute the clear operation.
+    pub fn prepare_clear_wallet(&self) -> Result<ClearWalletInfo> {
+        let amount_sat = self
+            .rt
+            .handle()
+            .block_on(self.sdk.max_reverse_swap_amount())
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Failed to get max reverse swap amount",
+            )?
+            .total_sat;
+
+        let reverse_swap_fees = self
+            .rt
+            .handle()
+            .block_on(self.sdk.fetch_reverse_swap_fees(ReverseSwapFeesRequest {
+                send_amount_sat: Some(amount_sat),
+            }))
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Failed to get max reverse swap amount",
+            )?;
+
+        let total_fees_sat = reverse_swap_fees
+            .total_estimated_fees
+            .ok_or_permanent_failure(
+                "No total reverse swap fee estimation provided when amount was present",
+            )?;
+        let onchain_fee_sat = reverse_swap_fees.fees_claim + reverse_swap_fees.fees_lockup;
+        let swap_fee_sat =
+            ((amount_sat as f64) * reverse_swap_fees.fees_percentage / 100_f64) as u64;
+        let exchange_rate = self.get_exchange_rate();
+
+        Ok(ClearWalletInfo {
+            clear_amount: amount_sat.as_sats().to_amount_up(&exchange_rate),
+            total_estimated_fees: total_fees_sat.as_sats().to_amount_up(&exchange_rate),
+            onchain_fee: onchain_fee_sat.as_sats().to_amount_up(&exchange_rate),
+            swap_fee: swap_fee_sat.as_sats().to_amount_up(&exchange_rate),
+            fees_hash: reverse_swap_fees.fees_hash,
+        })
+    }
+
+    /// Starts a reverse swap that sends all funds in LN channels to the provided onchain address.
+    pub fn clear_wallet(
```

Comment:
Mobile might be interested in an identifier of some sorts to navigate to the status page